glambie/data/timeseries.py

```diff
@@ -201,3 +212,381 @@ def metadata_as_dataframe(self) -> pd.DataFrame:
+                area_change = self.region.area_change
+                gt_adjusted_changes = []
+                for _, row in self.data.as_dataframe().iterrows():
+                    t_i = (row["start_dates"] + row["end_dates"]) / 2
+                    adjusted_area = glacier_area + (t_i - t_0) * (area_change / 100) * glacier_area
+                    gt_adjusted_changes.append(meters_water_equivalent_to_gigatonnes(
+                        [row.changes], area=adjusted_area, density_of_water=density_of_water)[0])
+                object_copy.data.changes = np.array(gt_adjusted_changes)
+                return object_copy
+        else:
+            raise NotImplementedError(
+                "Conversion to Gt not implemented yet for Timeseries with unit '{}'".format(self.unit))
+
+    def convert_timeseries_to_monthly_grid(self) -> Timeseries:
+        """
+        Converts a Timeseries object to follow the monthly grid. Two different approaches are used depending on
+        time resolution of the timeseries:
+
+        1.) Resolution over half a year: Timeseries is shifted to the closest month within the monthly grid
+            This method cannot be applied for monthly resolution as it may lead to empty months.
+        2.) Resolution under half a year: Timeseries is resampled to the monthly grid
+
+        Returns
+        -------
+        Timeseries
+            A copy of the Timeseries object containing the converted timeseries data to the monthly grid.
+        """
+        # make a deep copy of itself
+        object_copy = copy.deepcopy(self)
+        if not self.timeseries_is_monthly_grid():  # if already in monthly grid there is no need to convert
+            # check resolution
+            if self.data.max_temporal_resolution >= 0.5:  # resolution above half a year: shift to closest month
+                start_dates = timeseries_as_months(self.data.start_dates, downsample_to_month=False)
+                end_dates = timeseries_as_months(self.data.end_dates, downsample_to_month=False)
+                object_copy.data.start_dates = np.array(start_dates)
+                object_copy.data.end_dates = np.array(end_dates)
+            else:  # resolution below half a year: resample timeseries to monthly grid
+                start_dates, end_dates, changes = resample_derivative_timeseries_to_monthly_grid(self.data.start_dates,
+                                                                                                 self.data.end_dates,
+                                                                                                 self.data.changes)
+                object_copy.data.start_dates = np.array(start_dates)
+                object_copy.data.end_dates = np.array(end_dates)
+                object_copy.data.changes = np.array(changes)
+        return object_copy  # return copy of itself
+
+    def convert_timeseries_to_annual_trends(self, year_type="calendar") -> Timeseries:
```

Comment:
It might be better to create an enum for the year type.

Reply:
How do you mean exactly? Like this: https://docs.python.org/3/library/enum.html within the constants?

Reply:
Thanks, Created a YearType(Enum) in constants.py